modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/views/ViewsContext.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import React from 'react';
+
+import {ACTION_UPDATE_ACTIVE_VIEW} from '../actions/updateActiveView';
+import {ACTION_UPDATE_VIEW_COMPONENT} from '../actions/updateViewComponent';
+
+export const viewsReducer = (state, {type, value}) => {
+	const {activeView, views = []} = state;
```

Comment:
nit: It does seem a bit odd for the reducer to materialize an empty `views` array by default (ie. not in response to action). I'd rather we ensure that we always pass a `views` array into the `App` component. I see that's coming via display context something serializer something 😀, so I am not sure if it is an easy thing to guarantee or not. If you don't want to guarantee it at the server level, you could do it in `App` (in modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/entry.js) instead of in this file:

```js
const {activeViewName, views = []} = props;
```

Reply:
Yeah, makes sense.